<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1139</width>
    <height>496</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>780</width>
    <height>450</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>10000</width>
    <height>10000</height>
   </size>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::NoContextMenu</enum>
  </property>
  <property name="windowTitle">
   <string>Редактор графов</string>
  </property>
  <property name="windowIcon">
   <iconset theme="accessories-text-editor">
    <normaloff>.</normaloff>.</iconset>
  </property>
  <property name="inputMethodHints">
   <set>Qt::ImhLatinOnly</set>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonIconOnly</enum>
  </property>
  <property name="animated">
   <bool>true</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AnimatedDocks</set>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="button_delver">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>70</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>Удалить вершину</string>
    </property>
   </widget>
   <widget class="QFrame" name="weight">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>20</y>
      <width>71</width>
      <height>51</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QSpinBox" name="spin_dist">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>61</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>999</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_dist">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>49</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Вес</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="to">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>10</y>
      <width>161</width>
      <height>61</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_to">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>10</y>
       <width>49</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>В</string>
     </property>
    </widget>
    <widget class="QComboBox" name="combo_to">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>30</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <item>
      <property name="text">
       <string>a</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>b</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>c</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>d</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>e</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>g</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>h</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>f</string>
      </property>
     </item>
    </widget>
    <widget class="QLineEdit" name="edit_to">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>30</y>
       <width>113</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="for_2">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>10</y>
      <width>161</width>
      <height>61</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_for">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>10</y>
       <width>49</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Из</string>
     </property>
    </widget>
    <widget class="QComboBox" name="combo_for">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>30</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <item>
      <property name="text">
       <string>a</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>b</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>c</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>d</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>e</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>g</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>h</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>f</string>
      </property>
     </item>
    </widget>
    <widget class="QLineEdit" name="edit_for">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>30</y>
       <width>113</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label_commandline">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>270</y>
      <width>751</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>&gt;&gt; </string>
    </property>
   </widget>
   <widget class="QFrame" name="paste">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>311</width>
      <height>91</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_pastegraph">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>10</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Вставить граф</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="edit_pastegraph">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>30</y>
       <width>301</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="button_addgraph">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>60</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>Вставить</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="thisgraph">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>761</width>
      <height>141</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QTextEdit" name="edit_thisgraph">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>741</width>
       <height>51</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Panel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
     <property name="html">
      <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;justify&quot; style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_thisgraph">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Текущий граф</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_orient">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>10</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>Ориентированный</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="button_cleargraph">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>Очистить</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="button_addver">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>70</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>Добавить вершину</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="functions">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>300</y>
      <width>761</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Функции</string>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <widget class="QCheckBox" name="check_smej">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>141</width>
       <height>20</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>Матрица смежности</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_dost">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>151</width>
       <height>20</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>Матрица достижимости</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_vzaim">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>211</width>
       <height>20</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>Матрица взаимной достижимости</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_printgraph">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>20</y>
       <width>121</width>
       <height>20</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>Нарисовать граф</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_minostPrim">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>60</y>
       <width>181</width>
       <height>20</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>Мин. остовое дерево (Прим)</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_eiler">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>40</y>
       <width>91</width>
       <height>20</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>Эйлеровость</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_components">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>20</y>
       <width>201</width>
       <height>20</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>Компоненты сильной связности</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_dekstra">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>40</y>
       <width>201</width>
       <height>20</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>Кратчайшие пути из вершины</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_minostKruskal">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>60</y>
       <width>201</width>
       <height>20</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>Мин. остовное дерево (Крускал)</string>
     </property>
    </widget>
    <widget class="QComboBox" name="combo_vertices">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>60</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="matrix">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>10</y>
      <width>351</width>
      <height>431</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLineEdit" name="edit_currentTable">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>350</y>
       <width>281</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QTableWidget" name="table_main">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>331</width>
       <height>331</height>
      </rect>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
     <property name="dragEnabled">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QToolButton" name="button_changeTable">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>400</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&amp;gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="text">
      <string>&lt;-&gt;</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="edit_eiler">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>380</y>
       <width>281</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>10</y>
      <width>16</width>
      <height>91</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="button_func">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>410</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>Выполнить</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1139</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
